astropy/coordinates/transformations.py

```diff
@@ -741,8 +742,105 @@ def __call__(self, fromcoord, toframe):
                 'should have been of type {1}'.format(res, self.tosys))
         return res
 
+class BaseAffineTransform(CoordinateTransform):
+    """
+    Base class for common functionality between the ``AffineTransform``-type
+    subclasses.
+    """
+
+    def _apply_transform(self, fromcoord, matrix, vectors):
+        from .representation import (CartesianRepresentation,
+                                     UnitSphericalRepresentation,
+                                     CartesianDifferential)
+
+        rep = fromcoord.represent_as(CartesianRepresentation)
+
+        # Only do transform is matrix is specified. This is for speed in
+        # transformations that only specify an offset (e.g., LSR)
+        if matrix is not None:
+            newrep = rep.without_differentials().transform(matrix)
+        else:
+            newrep = rep
+
+        if vectors is not None and vectors[0] is not None:
+            pos_offset = CartesianRepresentation(vectors[0])
+            newrep = newrep + pos_offset
+
+        if rep.differentials:
+            # assume the 0th item is a velocity
+            # TODO: we only support velocities right now - elsewhere, we need to
+            # validate / make sure only 1 differential is attached
+            veldiff = rep.differentials[0]
+
+            base = fromcoord.data.represent_as(veldiff.base_representation)
+
+            # Only do transform is matrix is specified.
+            newdiff = veldiff.represent_as(CartesianRepresentation, base=base)
+            if matrix is not None:
+                newdiff = newdiff.transform(matrix)
+
+            if vectors is not None and vectors[1] is not None:
+                vel_offset = CartesianRepresentation(vectors[1])
+                newdiff = newdiff + vel_offset
+
+            newdiff = newdiff.represent_as(CartesianDifferential)
+            newrep._differentials = (newdiff,)
+
+        if issubclass(fromcoord.data.__class__, UnitSphericalRepresentation):
+            # need to special-case this because otherwise the new class will
+            # think it has a valid distance
+            newrep = newrep.represent_as(fromcoord.data.__class__)
 
-class StaticMatrixTransform(CoordinateTransform):
+        return newrep
+
+class AffineTransform(BaseAffineTransform):
```

Comment:
This (and probably `BaseAffineTransform`) should be in `__all__` , right?  That is, they shold be considered public API?